internal/provider/resources/deployment_test.go

```diff
@@ -101,33 +101,80 @@ resource "prefect_deployment" "{{.DeploymentName}}" {
 	work_queue_name = "{{.WorkQueueName}}"
 	parameter_openapi_schema = jsonencode({{.ParameterOpenAPISchema}})
 	pull_steps = [
-		{{range .PullSteps}}
+	  {{range .PullSteps}}
```

Comment:
The changes below are a bit hard to read, and repetitive. I tried importing `masterminds/sprig` to give us `coalesce` and some other helpers, but it started escaping quotations and I couldn't find any reference as to why that was happening. I ditched that to stick with the standard library's `template` and use `or`, which seemed to work fine at giving me a context based on which field wasn't empty - but then I was getting errors that it couldn't access nested fields when they were undefined.

This may actually be doable, but to get this PR rolling I just added a little duplication for the common fields and moved on.